### ISSUE TITLE
Accessibility for tooltips and backend

### ISSUE THREAD

**Issue by EugenioPetulla** [CodeAtCode/Glossary#35]
We have to figure out how to improve the accessibility for both, front and back end.

Maybe @afercia could point us to the right direction. :)

**Comment by Mte90** [comment]
I think that the backend is not a problem because is made it with CMB2 but the frontend require surely the aria-controls tags.

**Comment by EugenioPetulla** [comment]
Yes but I'm figuring out how screen readers and other stuff can understand the difference between a tooltip and another element inside the page. We made a sort of "hardcode" into the content hided by CSS. A simple `<span>` element within a class is not a good starting point for setting up a clear difference between main text and extra info.

**Comment by afercia** [comment]
Hello :) Accessible tooltips are a bit challenging, by the way if you look even here on GitHub when you hover on the notifications "bell" or your profile icon on the screen top right, you will see a tooltip. They appear also on focus, when using the keyboard. The tooltip content comes from an `aria-label` attribute. Building a tooltip would be just a few lines of CSS using generated content and the `aria-label` attribute as value. This could be a start, or at least something to try.
I guess a proper tooltip positioning would always require some JS.

As far as I see the extension injects some HTML, you could try using an `aria-label` attribute on this span:
```
<span class="glossary-word-glotdict" data-translations="[{&quot;translation&quot;:&quot;profilo&quot;,&quot;pos&quot;:&quot;&quot;,&quot;comment&quot;:&quot;&quot;}]" style="cursor: help; border-bottom-width: 1px; border-bottom-style: dashed;">profile</span>
```

- worth considering `aria-label` attributes completely override the element content so the strings to use as values should be carefully crafted and I'm not so sure in complex, long, sentences they would make sense
- for keyboard users (no screen reader) the element should be focusable

TL;DR more details on Slack?

**Comment by EugenioPetulla** [comment]
@afercia this is a different glossary... we develop to many and now you are confusing projects XD

This is our Glossary plugin for WordPress and it's all about PHP rendering HTML directly into the page.

Look here for the code: https://github.com/CodeAtCode/Glossary

And here is the WP repo link: https://wordpress.org/plugins/glossary-by-codeat/

**Comment by afercia** [comment]
Yep sorry, took me a while to understand we were talking about two different things. My bad :)
